### PR TITLE
[build] Update a bunch of "useless casts", make compiler happy

### PR DESCRIPTION
**Pull request by WinterSolstice8**
* Note, this commit doesn't fix the build entirely for GCC
dependencies still error. Even when build is successful, topaz_game crashes on login.
Please scrutinize this commit to the furthest extent possible.

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [☠️] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

File: src/map/packet_system.cpp [hunk 1]
```diff
@@ -3825,7 +3827,12 @@ void SmallPacket0x064(map_session_data_t* const PSession, CCharEntity* const PCh
         return;
     }
 
-    memcpy(&PChar->keys.tables[KeyTable].seenList, data[0x08], 0x40);
+    // memcpy(&PChar->keys.tables[KeyTable].seenList, data[0x08], 0x40);
+    // TODO: verify this works?
+    for (int i = 0; i < 0x40; i++)
+    {
+        PChar->keys.tables[KeyTable].seenList[i] = data[0x08 + i];
```
Comment (WinterSolstice8) [inline src/map/packet_system.cpp:3834]: Please double check this. No idea if this is the correct syntax, memset specifically was called out as bad to use against std::bitset by gcc.

Comment (zach2good) [reply src/map/packet_system.cpp:3834]: https://stackoverflow.com/questions/18513259/how-to-initialize-a-bitset-type-all-bits-to-1

Looks like `set` is a good option here for `bitset`

Comment (WinterSolstice8) [reply src/map/packet_system.cpp:3834]: Looking at it more, memset here was copying 64 bytes and this for loop is only setting 64 bits. So this implementation is definitely wrong

**Review by zach2good — APPROVED** [review]
This is good work, I'm going to merge this in (since it's going into my other working branch, and it'll need a lot more work anyway)!

**Pull request merged**